File: 김찬우/프로그래머스 /Lv.1/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        
        PriorityQueue<Integer> pQ = new PriorityQueue<>(Collections.reverseOrder());
        
        for(int i=0; i<priorities.length; i++){
            pQ.offer(priorities[i]);
        }
        
        while(!pQ.isEmpty()){
            
            for(int i=0; i<priorities.length; i++){
                
                if(pQ.peek() == priorities[i]){
                    pQ.poll();
                    answer++;
                    
                    if(i == location){
                        return answer;
                    }
                    
                }
                
            }
            
        }
        
        
        
        
        return answer;
    }
}

File: 조준장/level2/프로세스.kt
class Solution {
    fun solution(priorities: IntArray, location: Int): Int {
        var answer = 0
        var newPriorities : IntArray = priorities
        var target = newPriorities.maxOf{it}
        
        while (true){
            newPriorities.indices.forEach{
                
                if (newPriorities[it] == 0) return@forEach
                else if (newPriorities[it] == target) {
                    answer ++
                    if (it == location) return@solution answer
                    newPriorities[it] = 0
                    target = newPriorities.maxOf{it}  
                }
            }
        }
        
        return answer
    }
}
